src/test/fuzz/base_encode_decode.cpp

```diff
@@ -6,49 +6,92 @@
 
 #include <base58.h>
 #include <psbt.h>
+#include <test/fuzz/FuzzedDataProvider.h>
 #include <util/strencodings.h>
 #include <util/string.h>
 
 #include <cassert>
-#include <cstdint>
 #include <string>
 #include <vector>
+#include <ranges>
 
-using util::TrimString;
 using util::TrimStringView;
 
-FUZZ_TARGET(base_encode_decode)
+FUZZ_TARGET(base58_encode_decode)
 {
-    const std::string random_encoded_string(buffer.begin(), buffer.end());
+    FuzzedDataProvider provider(buffer.data(), buffer.size());
+    const std::string random_string{provider.ConsumeRandomLengthString(1000)};
+    const auto max_ret_len{provider.ConsumeIntegralInRange<int>(1, 300)};
```

Comment:
nit: As `random_string` is currently `0-1000` long, and `max_ret_len` is `1-300`, most of the time `DecodeBase58()` is just returning `false` since `max_ret_len` is too small. But maybe that is intentional to stress that aspect?

Might be better to have more similar lengths, and also to include `0` as `max_ret_len` since it might occur in places where that parameter is calculated.

```suggestion
    const std::string random_string{provider.ConsumeRandomLengthString(1000)};
    const auto max_ret_len{provider.ConsumeIntegralInRange<int>(0, 1000)};
```

Reply:
Good point, and even negative values theoretically - thanks, [pushed](https://github.com/bitcoin/bitcoin/compare/a32002f2d5cff72639c9782d70ae52ec162d9ef8..6983d82c84d4ca351a7cd9d1e0e20ab878da6475)